jwst/source_catalog/source_catalog.py

```diff
@@ -84,6 +84,9 @@ def make_source_catalog(model, kernel_fwhm, kernel_xsize, kernel_ysize,
     if not isinstance(model, DrizProductModel):
         raise ValueError('The input model must be a DrizProductModel.')
 
+    # Mask the non-imaging area (e.g. MIRI)
+    mask = (model.dqflags.pixel['NON_SCIENCE'] & model.dq).astype(np.bool)
```

Comment:
There is no `dq` array in the `resample` output unfortunately.  And in fact, the resampled output should have already taken into account the `'NON_SCIENCE'` flagging and not resampled those pixels to the output.  I suspect that all the changes in this PR to `source_catalog` can be removed, and only those to `tweakreg` will be necessary.

Reply:
The only time a DQ array would be present is if the image has not been resample (i.e. not a DrizProductModel). Could something like a weight or context value of 0 in the DrizProduct serve a similar purpose?